alf/algorithms/rl_algorithm.py

```diff
@@ -187,6 +187,17 @@ def action_spec(self):
         """Return the action spec."""
         return self._action_spec
 
+    @property
+    def train_action_spec(self):
```

Comment:
I think the issue is that summarize_train is always called. And it will assume the action spec is the original action spec in https://github.com/HorizonRobotics/alf/blob/pytorch/alf/algorithms/rl_algorithm.py#L262 . At least we need an option to disable the default summarize_train.